### ISSUE TITLE
[CSV][GQLDF] If you delete CSV rows, and update raw, you get duplicate RAW rows...

### ISSUE THREAD

**Issue by idiom-bytes** [oceanprotocol/pdr-backend#1087]
### Issue: CSV Deletion/Raw Table Duplication 

**TLDR; just drop raw and etl tables before updating and getting new CSVs.**

Checks in save_to_storage against duckdb raw tables

The reason the problem above happened is that GQLDF uses save_to_storage, which doesn't do any checks against the DuckDB table when doing the insert. This is because it assumes that the DuckDB table will be in the same state as the CSV... clean, empty.

So, it's currently the responsibility of the user, that if they delete the CSV, that they'll also trim the DuckDB table (lake) with the drop commands that are currently there.

And, let's compartmentalise the problem...
- The whole data pipeline works on the assumption that users will be iterating and dropping their database tables (DuckDB) to rebuild from scratch using the CSVs, with frequency.
**- Right now, we do not want users to delete their CSV, or expose them to managing this.**
- The whole data pipeline works on the assumption that they only have to fetch this data once. Read: low touch/velocity.

**This is a low priority issue**

### [CSV Management - Considerations / Potential Solutions]
- Keep track of table min/max timestamp.
- Checks are made in GQLDF -> append_to_table -> insert_to_duckdb (checks the row to be inserted is less than min or greater than max). This way, we update CSV but don't insert double records into lake.
- We continue to extend CI commands in such a way, where you can also manage the CSV data easily.
- We continue to extend CI commands in such a way, where you can drop across all CSV/RAW/ETL easily.
- Even right now, if you do delete some of your CSVs, you can get your tables up and running again in just a few seconds.

**Comment by kdetry** [comment]
I have following ideas as a potential solution:

Unique Constraints in DuckDB:
Adding unique constraints (ID) in DuckDB to enforce data integrity. This will automatically prevent duplicate entries from being inserted into the table.
Track the last update point and only append new records beyond that point to both the CSV and DuckDB table.

Cleanup Scripts:
For whole csv/raw/etl processes, we could create cleaner drop scripts

**Comment by calina-c** [comment]
Why complicate with timestamps or ids? Can't we do a simple differential of identical rows? Before saving/updating from csv, remove rows that already exist in the table.

**Comment by idiom-bytes** [comment]
```
Adding unique constraints (ID) in DuckDB to enforce data integrity. This will automatically prevent duplicate entries from being inserted into the table.`
```

```
Why complicate with timestamps or ids? Can't we do a simple differential of identical rows? Before saving/updating from csv, remove rows that already exist in the table.
```

You guys are advocating for vanilla fetch and join strategies that lead to N<>N joins that have O(n^2). Please go ahead and try doing this with 1 million rows of anything. Further, these checks are constant, and across the whole database... rather then just using checkpoints (i.e. a timestamp) so you can resume the workflow.

There is a reason why we've worked on this for months, and it's to build a pipeline that processes records efficiently, does as few scans and joins as possible, and it can scale.

**Comment by idiom-bytes** [comment]
This is being tracked in the backlog. Closing for now, reopen 1 by 1.
